### DISCUSSION TITLE
Array class for initExe in Delphi

### DISCUSSION THREAD

**Discussion by 2A2U** [Vector35/binaryninja-api#5397] [Q&A] [answered]
Hi People, 

For context, I was doing reverse on a Delphi program because I wanted to try it by using a Binary Ninja. However, I encountered an issue where Binary Ninja was recognizing bytes of double dword as a function, as depicted in the image below:
![image](https://github.com/Vector35/binaryninja-api/assets/121478081/59e13031-11ef-4378-9d3a-2fecc90655ee)

Whereas, when I attempted the same analysis using IDA, it successfully recognized the types, as shown in the following screenshot:
![Screenshot_3](https://github.com/Vector35/binaryninja-api/assets/121478081/fecad2b1-b494-4ce6-8b1d-29d7fc1f9341)

I tried to undefine the function and manually change it to double dword. However, it takes time, as it requires modification per 4 bytes. Is there a more efficient way to achieve the same result as IDA in Binary Ninja Free version?

sample is at [Malware Bazaar](https://bazaar.abuse.ch/browse.php?search=sha256%3A8be8abab0f3ebcc18e7b8b3c437feb073fa7b9b30e8b1eef81451f0938b2a52a) and thanks in advance

**Comment by plafosse** [comment, marked as answer]
Unfortunately we don't have great support for delphi. One option would be to reduce the number of bad functions is to disable linearsweep when you "Open with options" To create a bunch of dwords you can create a selection and hit `4`

**Reply by xusheng6** [reply]
yeah, in other words, you do not need to create these dwords one-by-one -- just select the entire range, and press "4", binja will create all of the dwords all at once
